src/common/snippets/src/pass/collapse_subgraph.cpp

```diff
@@ -611,7 +611,14 @@ TokenizeSnippets::TokenizeSnippets() {
         // This limitation will be resolved once generator supports gprs spills [75622].
         // TODO [75567]: move this plugin-specific constraint to the plugin callback
         const auto unique_buffer_count = op::Subgraph::get_estimated_buffer_count(ops_for_buffer_count);
-        if (body_parameters.size() + body_results.size() + hidden_data_count + unique_buffer_count > 11) {
+#if defined(OPENVINO_ARCH_ARM64)
+        // ARM has 32 gprs. After excluding 2 registers for work amounts, 1 platform register, 3 registers for temporary use,
+        // and two stack related registers, it has 24 remaining registers.
+        const size_t max_data_ptr_count = 24;
+#else
+        const size_t max_data_ptr_count = 11;
+#endif
+        if (body_parameters.size() + body_results.size() + hidden_data_count + unique_buffer_count > max_data_ptr_count) {
```

Comment:
@a-sidorova, please note that we have [tokenization_config](https://github.com/openvinotoolkit/openvino/blob/fd88a6b4551da5fdb17064037b400dc130379f6f/src/common/snippets/include/snippets/pass/tokenization.hpp#L68-L84) that can be useful here. We can pass the max number of registers through the config, this will allow to remove the arch-specific define from the common part. What do you think?

Reply:
@IvanNovoselov good idea! We can adjust tokenization using this config. Thanks!

@xuchen-intel could you please add the separate field in tokenization config and init with arch-dependent value?

Reply:
Applied. Thanks Alexandra and Ivan!